mlflow/store/db/utils.py

```diff
@@ -127,32 +127,25 @@ def _get_managed_session_maker(SessionMaker, db_type):
     @contextmanager
     def make_managed_session():
         """Provide a transactional scope around a series of operations."""
-        session = SessionMaker()
-        try:
-            if db_type == SQLITE:
-                session.execute(sql.text("PRAGMA foreign_keys = ON;"))
-                session.execute(sql.text("PRAGMA busy_timeout = 20000;"))
-                session.execute(sql.text("PRAGMA case_sensitive_like = true;"))
-            yield session
-            session.commit()
-        except MlflowException:
-            session.rollback()
-            raise
-        except sqlalchemy.exc.OperationalError as e:
-            session.rollback()
-            _logger.exception(
-                "SQLAlchemy database error. The following exception is caught.\n%s",
-                e,
-            )
-            raise MlflowException(message=e, error_code=TEMPORARILY_UNAVAILABLE)
-        except sqlalchemy.exc.SQLAlchemyError as e:
-            session.rollback()
-            raise MlflowException(message=e, error_code=BAD_REQUEST)
-        except Exception as e:
-            session.rollback()
-            raise MlflowException(message=e, error_code=INTERNAL_ERROR)
-        finally:
-            session.close()
+        with SessionMaker.begin() as session:
+            try:
+                if db_type == SQLITE:
+                    session.execute(sql.text("PRAGMA foreign_keys = ON;"))
+                    session.execute(sql.text("PRAGMA busy_timeout = 20000;"))
+                    session.execute(sql.text("PRAGMA case_sensitive_like = true;"))
+                yield session
+            except sqlalchemy.exc.OperationalError as e:
+                _logger.exception(
+                    "SQLAlchemy database error. The following exception is caught.\n%s",
+                    e,
+                )
+                raise MlflowException(message=e, error_code=TEMPORARILY_UNAVAILABLE)
+            except sqlalchemy.exc.SQLAlchemyError as e:
+                raise MlflowException(message=e, error_code=BAD_REQUEST)
+            except MlflowException:
+                raise
+            except Exception as e:
+                raise MlflowException(message=e, error_code=INTERNAL_ERROR)
```

Comment:
/ai This is the code to manage a session in sqlalchemy. Please review it and tell me issues if you find any.